app_window.py

```diff
@@ -334,29 +335,56 @@ def __init__(self, *args, **kwargs):
                 "toggle through tags directories at any time with F5.",
                 parent=self)
 
+    @property
+    def last_data_load_dir(self):
+        return self._last_data_load_dir
+    @last_data_load_dir.setter
+    def last_data_load_dir(self, new_val):
+        if not isinstance(new_val, Path):
+            new_val = Path(new_val)
+        self._last_data_load_dir = new_val
+
+    @property
+    def jms_load_dir(self):
+        return self._jms_load_dir
+    @jms_load_dir.setter
+    def jms_load_dir(self, new_val):
+        if not isinstance(new_val, Path):
+            new_val = Path(new_val)
+        self._jms_load_dir = new_val
+
+    @property
+    def bitmap_load_dir(self):
+        return self._bitmap_load_dir
+    @bitmap_load_dir.setter
+    def bitmap_load_dir(self, new_val):
+        if not isinstance(new_val, Path):
+            new_val = Path(new_val)
+        self._bitmap_load_dir = new_val
+
     @property
     def data_dir(self):
         tags_dir = self.tags_dir
-        if not tags_dir:
+        if tags_dir is None:
             return e_c.WORKING_DIR
 
         return path_replace(tags_dir, "tags", "data")
 
     @property
     def tags_dir(self):
         try:
-            return self.tags_dirs[self._curr_tags_dir_index]
+            return Path(self.tags_dirs[self._curr_tags_dir_index])
         except IndexError:
             return None
 
     @tags_dir.setter
     def tags_dir(self, new_val):
         assert isinstance(new_val, (str, Path))
-        self.tags_dirs[self._curr_tags_dir_index] = path_normalize(new_val)
+        self.tags_dirs[self._curr_tags_dir_index] = Path(path_normalize(new_val))

```

Comment:
You're double dipping. Path already normalizes.
```suggestion
        self.tags_dirs[self._curr_tags_dir_index] = Path(new_val)
```